5.5/root/usr/share/container-scripts/mysql/validate-variables.sh

```diff
@@ -32,11 +30,34 @@ function validate_variables() {
     root_specified=1
   fi
 
+  # If MYSQL_USER == "root", we have a special case
+  if [[ "${user_specified:-0}" == "1" && "$MYSQL_USER" == "root" ]]; then
+    if [[ "${root_specified:-0}" == "1" ]]; then
+      usage "You cannot set MYSQL_USER to 'root' and use MYSQL_PASSWORD and MYSQL_ROOT_PASSWORD at the same time"
```

Comment:
This solution would make sense to me as well, although I'm not entirely convinced whether one of the solution is better than the other; this one would be simpler to implement at least.